File: graphDFS.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n, m;
    cin >> n >> m;

    vector<vector<int>> adj(n + 1, vector<int>(n + 1, 0));

    for (int i = 0; i < m; i++) {
        int u, v;
        cin >> u >> v;
        adj[u][v] = 1;
        adj[v][u] = 1;  
    }
    int start;
    cout << "Start node: ";
    cin >> start;

    vector<bool> visited(n + 1, false);
    stack<int> s;

    s.push(start);

    cout << "DFS " << start << ": ";

    while (!s.empty()) {
        int node = s.top();
        s.pop();

        if (!visited[node]) {
            cout << node << " ";
            visited[node] = true;

            for (int neighbor = n; neighbor >= 1; neighbor--) {
                if (adj[node][neighbor] == 1 && !visited[neighbor]) {
                    s.push(neighbor);
                }
            }
        }
    }
    cout << "\n";
    return 0;
}

File: graphDFS.py
def dfs(n, m, edges, start):
    adj = [[0] * (n + 1) for _ in range(n + 1)]

    for u, v in edges:
        adj[u][v] = 1
        adj[v][u] = 1

    visited = [False] * (n + 1)
    stack = []

    stack.append(start)

    print(f"DFS {start}: ", end="")

    while stack:
        node = stack.pop()

        if not visited[node]:
            print(node, end=" ")
            visited[node] = True

            for neighbor in range(n, 0, -1):
                if adj[node][neighbor] == 1 and not visited[neighbor]:
                    stack.append(neighbor)
    print()

if __name__ == "__main__":
    n, m = map(int, input().split())
    edges = [tuple(map(int, input().split())) for _ in range(m)]
    start = int(input("Start node: "))
    dfs(n, m, edges, start)
